modular_citadel/code/modules/arousal/organs/penis.dm

```diff
@@ -18,23 +18,56 @@
 	var/knot_girth_ratio 	= KNOT_GIRTH_RATIO_DEF
 	var/list/dickflags 		= list()
 	var/list/knotted_types 	= list("knotted", "barbed, knotted")
+	var/prev_size			= 6 //really should be renamed to prev_length
+
+/obj/item/organ/genital/penis/Initialize()
+	. = ..()
+	/* I hate genitals.*/
 
 /obj/item/organ/genital/penis/update_size()
-	if(length == cached_length)
+	var/mob/living/carbon/human/o = owner
+	if(!ishuman(o) || !o)
 		return
-	switch(length)
-		if(-INFINITY to 5)
+	if(cached_length < 0)//I don't actually know what round() does to negative numbers, so to be safe!!
+		var/obj/item/organ/genital/penis/P = o.getorganslot("penis")
+		to_chat(o, "<span class='warning'>You feel your tallywacker shrinking away from your body as your groin flattens out!</b></span>")
+		P.Remove(o)
+	switch(round(cached_length))
+		if(0 to 4) //If modest size
+			length = cached_length
 			size = 1
-		if(5 to 9)
+			if(owner.has_status_effect(/datum/status_effect/chem/PElarger))
+				o.remove_status_effect(/datum/status_effect/chem/PElarger)
+		if(5 to 10) //If modest size
+			length = cached_length
 			size = 2
-		if(15 to INFINITY)
-			size = 3//no new sprites for anything larger yet
-/*		if(9 to 15)
+			if(owner.has_status_effect(/datum/status_effect/chem/PElarger))
+				o.remove_status_effect(/datum/status_effect/chem/PElarger)
+		if(11 to 20) //If massive
+			length = cached_length
+			size = 3
+			if(owner.has_status_effect(/datum/status_effect/chem/PElarger))
+				o.remove_status_effect(/datum/status_effect/chem/PElarger)
+		if(21 to 35) //If massive and due for large effects
+			length = cached_length
 			size = 3
-		if(15 to INFINITY)
-			size = 3*/
-	girth = (length * girth_ratio)
-	cached_length = length
+			if(!owner.has_status_effect(/datum/status_effect/chem/PElarger))
+				o.apply_status_effect(/datum/status_effect/chem/PElarger)
+		if(36 to INFINITY) //If comical
+			length = cached_length
+			size = 4 //no new sprites for anything larger yet
+			if(!owner.has_status_effect(/datum/status_effect/chem/PElarger))
+				o.apply_status_effect(/datum/status_effect/chem/PElarger)
+
+	if (round(length) > round(prev_size))
+		to_chat(o, "<span class='warning'>Your [pick("phallus", "willy", "dick", "prick", "member", "tool", "gentleman's organ", "cock", "wang", "knob", "dong", "joystick", "pecker", "johnson", "weenie", "tadger", "schlong", "thirsty ferret", "baloney pony", "schlanger")] [pick("swells up to", "flourishes into", "expands into", "bursts forth into", "grows eagerly into", "amplifys into")] a [uppertext(round(length))] inch penis.</b></span>")
```

Comment:
I feel like these size memes for all organs could be made into set global lists to be chosen from, instead of picked each time to_chat is called.